### PR TITLE
Backports/60x/v1

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/6403

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16162

File: src/tm-threads.c [hunk 1]
```diff
@@ -123,22 +140,11 @@ TmEcode TmThreadsSlotVarRun(ThreadVars *tv, Packet *p, TmSlot *slot)
...
-
-        /* handle new packets */
-        while (tv->decode_pq.top != NULL) {
-            Packet *extra_p = PacketDequeueNoLock(&tv->decode_pq);
-            if (unlikely(extra_p == NULL))
-                continue;
-
-            /* see if we need to process the packet */
-            if (s->slot_next != NULL) {
-                r = TmThreadsSlotVarRun(tv, extra_p, s->slot_next);
```
Comment (jufajardini) [inline src/tm-threads.c:135]: Was this bit connected to the behavior described in the commit for bug 6402? (Just curious)

Comment (victorjulien) [reply src/tm-threads.c:135]: it's a cherry-pick to make the backport a bit easier/cleaner

Comment (jufajardini) [reply src/tm-threads.c:135]: Ah, from the other description, I got curious if this part was also related to that. Thanks :)

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16163

**Review by jufajardini — APPROVED** [review]
LGTM!

**Comment by victorjulien** [comment]
Merged in #9643, thanks!

**Pull request closed without merging**